Automattic/wp-calypso
# Initial data layer and data query component for `/all-domains` endpoint

#### Changes proposed in this Pull Request

*  Gets domains data from a new HTTP path  i.e. `/all-domains`. 
*  A new data component `QueryAllDomains` to manage fetching domains data. 

#### Testing instructions



* Start your Calypso dev server and navigate to : http://calypso.localhost:3000/domains/manage
* Use the Redux profiler to ensure that a new key `allDomains` exists in the tree.
